File: UI/src/app/_model/Transaction.ts
import {Tag} from "./Tag";
import {Consumer} from "./Consumer";
import {TransactionInfo} from "./TransactionInfo";
import {TransactionType} from "./TransactionType";

export class Transaction {
    id!: string;
    transactionDate!: string;
    transactionInfo!: TransactionInfo;
    transactionType!: TransactionType;
    transactionNotice!: string;
    consumer!: Consumer;
    tags!: Tag[];
    total!: string;
}

File: handler/src/main/java/org/myproject/cash/cockpit/handler/model/Transaction.java
package org.myproject.cash.cockpit.handler.model;

import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder(toBuilder = true)
@EqualsAndHashCode
@RequiredArgsConstructor
public class Transaction {

    @NotNull
    @Builder.Default
    private final LocalDate transactionDate = LocalDate.now();

    @EqualsAndHashCode.Exclude
    @Builder.Default
    private final TransactionInfo transactionInfo = TransactionInfo.builder().build();

    @Builder.Default
    private final TransactionType transactionType = TransactionType.builder().build();

    @Builder.Default
    private final Sender sender = Sender.builder().build();

    @Builder.Default
    private final Destination destination = Destination.builder().build();

    @Builder.Default
    @EqualsAndHashCode.Exclude
    private final List<Tag> tags = new ArrayList<>();

    @NotNull
    private final double total;

}
